### ISSUE TITLE
Razer-macos 0.4.1

### ISSUE THREAD

**Issue by yabba235** [1kc/razer-macos#356]
Work under Razer Blade Base 2020 but do nothing - device list is empty.
Version master not work - java issue.

**Comment by WrathofthePast** [comment]
#354 I knew I wasn't crazy... I won't reopen my issue since this one is here now. I'll continue to look for the error log & upload it here.

**Comment by dhobi** [comment]
Are you talking about "Blade 15 Base (Early 2020)" (1532:0255)?
If so, could you change the title to something like "Support Blade 15 Base (Early 2020)"

Changes done in openrazer we would have to "replicate":
https://github.com/openrazer/openrazer/pull/1171

**Comment by WrathofthePast** [comment]
<img width="250" alt="Screen Shot 2021-04-24 at 12 08 17 AM" src="https://user-images.githubusercontent.com/75559873/116176750-ad946200-a6e0-11eb-975b-0de147e99915.png">
<img width="328" alt="Screen Shot 2021-04-24 at 12 08 29 AM" src="https://user-images.githubusercontent.com/75559873/116176759-b1c07f80-a6e0-11eb-8637-5439a877c6f1.png">

I managed to get the JavaScript Error Log. Please note I do not have a Blade, so this is pertains to either the program as a whole or my Deathadder v2 and/or Core X Chroma.

**Comment by LuxuryWasTaken** [comment]
guys fund the issue

**Comment by LuxuryWasTaken** [comment]
okey I added the support of 2020 Base according to this:
https://github.com/1kc/razer-macos/pull/295

Here is download link (didn't updated the source code)
https://github.com/LuxuryWasTaken/razer-macos/releases/tag/v0.4.2.1

Known issues: 
In the color selection gui (device) max brightness is like 30%, I'm increasing It manually (same for menu bar icon too)
Gui's color picker is not working.

But the reason I did that is because I don't like to use macOS with rgb, the whole point of this for me is to set color to Static White which is working by menu icon.

**Comment by dhobi** [comment]
> Known issues:
In the color selection gui (device) max brightness is like 30%, I'm increasing It manually (same for menu bar icon too)
Gui's color picker is not working.

It would be helpful to see your source code changes here to give any hints on what could be (possibly) improved. Nevertheless, great that you made it work for you! It's for sure a step in the right direction! 😃

**Comment by LuxuryWasTaken** [comment]
I really don’t know about coding, just checked pull request #295 added the lines in there to same files. So basically, in this request, dev is adding hunstman mini to some .c files. I went to those files, searched for blade_2019 (for reference, in order to add blade_2020 to correct category.) keep added blade_base_2020 to under of it. Then in a file, it requires a number starts with 0x I guess it’s device ID. Went to openrazer’s github, get this ID from there. Then created a json file in devices folder (copy of blade 2019 base [edited for 2020]) then compiled it with yarn.

basically those was my steps, I can post the modified files if you want.

**Comment by 1kc** [comment]
Could someone please report back on how well this new commit works? If not I'm trusting @dhobi's work 😃 and merging and releasing.

**Comment by WrathofthePast** [comment]
@1kc I've haven't had the JavaScript Error on v0.4.3-test, but I don't have a Blade 15 Base & still don't understand how these are related. I'm happy the JavaScript error screenshots I provided helped though.

**Comment by 1kc** [comment]
> @1kc I've haven't had the JavaScript Error on v0.4.3-test, but I don't have a Blade 15 Base & still don't understand how these are related. I'm happy the JavaScript error screenshots I provided helped though.

I think I might have fixed that error at some point, can't remember. If it doesn't show up thats good!

**Comment by WrathofthePast** [comment]
> > @1kc I've haven't had the JavaScript Error on v0.4.3-test, but I don't have a Blade 15 Base & still don't understand how these are related. I'm happy the JavaScript error screenshots I provided helped though.
> 
> I think I might have fixed that error at some point, can't remember. If it doesn't show up thats good!

It still happens in v0.4.2.

> Work under Razer Blade Base 2020 but do nothing - device list is empty.
> Version master not work - java issue.

The original description had mentioned a JavaScript issue.

> <img alt="Screen Shot 2021-04-24 at 12 08 17 AM" width="250" src="https://user-images.githubusercontent.com/75559873/116176750-ad946200-a6e0-11eb-975b-0de147e99915.png">
> 
> <img alt="Screen Shot 2021-04-24 at 12 08 29 AM" width="328" src="https://user-images.githubusercontent.com/75559873/116176759-b1c07f80-a6e0-11eb-8637-5439a877c6f1.png">
> 
> I managed to get the JavaScript Error Log. Please note I do not have a Blade, so this is pertains to either the program as a whole or my Deathadder v2 and/or Core X Chroma. Please let me know if my original issue should be reopened since this appears to be unrelated to Device Support.

Upon posting these screenshots of the JavaScript error here, this issue gathered immediate attention. The JavaScript error never impacted me & only appeared when restarting for a MacOS update. To me, it's seems like this issues were tied somehow despite that making no logical sense to me. I had opened an issue #354 which you left an emoji on. I eventually closed it when I thought I was imagining things. 

I mention all of this because I wanted you to know that I wasn't crazy. 

Note: I'm not saying you didn't fix it. I have no idea who did, but if you remember that, then you fixed it.
